internal/worker/caasapplicationprovisioner/ops.go

```diff
@@ -280,13 +279,7 @@ func appAlive(ctx context.Context, appName string, appUUID coreapplication.UUID,
 	storageUniqueID := getStorageUniqueID(appUUID)
 	filesystems := []internalstorage.KubernetesFilesystemParams{}
 	for _, fst := range pi.FilesystemTemplates {
-		for i := range fst.Count {
-			mountPoint, err := storage.FilesystemMountPointK8s(
-				fst.Location, fst.MaxCount, i, fst.StorageName,
-			)
-			if err != nil {
-				return errors.Trace(err)
-			}
+		for _, mountPoint := range fst.MountPoints {
```

Comment:
Hi Tom, from the PR message, I gathered that the main functional change here applies to IaaS deployments and a bug issue is created to track the missing change for Kubernetes. Could you help me understand why this change is made in the CAAS application provisioner worker if that’s the case?

Reply:
The bug that exist in the Kubernetes case is that we can't support multiple attachments for the same charm storage even though our model allows it. This is because of a logic error in the Kubernetes provider. We can fix this and it shouldn't be that hard I am just not tackling it at the moment because it exists in 3.6.

I have updated the logic so that the IAAS and CAAS both share the same logic. The new way allows the storage provisioning domain to drive more of the attachment logic and take control away from the provider(s).